File: VB/WinSolution.Module.Win/SuppressConfirmationForNewObjectsDetailViewController.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Win.SystemModule

Namespace WinSolution.Module.Win
	Public Class SuppressConfirmationForNewObjectsDetailViewController
		Inherits ViewController(Of DetailView)
		Private worker As WinDetailViewController
		Protected Overrides Overloads Sub OnActivated()
			MyBase.OnActivated()
			AddHandler ObjectSpace.ObjectChanged, AddressOf ObjectSpace_ObjectChanged
			If ObjectSpace.IsNewObject(View.CurrentObject) Then
				worker = Frame.GetController(Of WinDetailViewController)()
				worker.SuppressConfirmation = True
			End If
		End Sub
		Protected Overrides Overloads Sub OnDeactivating()
			RemoveHandler ObjectSpace.ObjectChanged, AddressOf ObjectSpace_ObjectChanged
			MyBase.OnDeactivated()
		End Sub
		Private Sub ObjectSpace_ObjectChanged(ByVal sender As Object, ByVal e As ObjectChangedEventArgs)
			If worker IsNot Nothing Then
				worker.SuppressConfirmation = False
			End If
		End Sub
	End Class
End Namespace

File: CS/WinSolution.Module.Win/SuppressConfirmationForNewObjectsDetailViewController.cs
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Win.SystemModule;

namespace WinSolution.Module.Win {
    public class SuppressConfirmationForNewObjectsDetailViewController : ViewController<DetailView> {
        private WinDetailViewController worker;
        protected override void OnActivated() {
            base.OnActivated();
            ObjectSpace.ObjectChanged += ObjectSpace_ObjectChanged;
            if (ObjectSpace.IsNewObject(View.CurrentObject)) {
                worker = Frame.GetController<WinDetailViewController>();
                worker.SuppressConfirmation = true;
            }
        }
        protected override void OnDeactivating() {
            ObjectSpace.ObjectChanged -= ObjectSpace_ObjectChanged;
            base.OnDeactivating();
        }
        private void ObjectSpace_ObjectChanged(object sender, ObjectChangedEventArgs e) {
            if (worker != null)
                worker.SuppressConfirmation = false;
        }
    }
}